tests/assembly/indexing-with-bools-no-redundant-instructions.rs

```diff
@@ -0,0 +1,30 @@
+//@ assembly-output: emit-asm
+//@ only-x86_64
+//@ compile-flags: -O
+//! Ensure that indexing a slice with `bool` does not
+//! generate any redundant `jmp` and `and` instructions.
+//! Discovered in issue #123216.
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+fn f(a: u32, b: bool, c: bool, d: &mut [u128; 2]) {
+    // CHECK-LABEL: f:
+    // CHECK: testl  %esi, %esi
+    // CHECK: je
+    // CHECK: xorb   %dl, %dil
+    // CHECK: orb    $1, (%rcx)
+    // CHECK: movzbl %dil, %eax
+    // CHECK: andl   $1, %eax
+    // CHECK: shll   $4, %eax
+    // CHECK: orb    $1, (%rcx,%rax)
+    // CHECK-NOT:     jmp
+    // CHECK-NOT:     andl %dil, $1
+    // CHECK: retq
```

Comment:
The `x86_64-fortanix-unknown-sgx` target applies LVI mitigations. This implies that instructions like `retq` must not be emitted. This makes this test incompatible with this target. Can you annotate it with:
```
//@ ignore-sgx Tests incompatible with LVI mitigations
```
Thanks!

Reply:
done, thanks!